python/lsst/daf/butler/registry/collections/_base.py

```diff
@@ -419,11 +415,13 @@ def getParentChains(self, key: Any) -> Iterator[ChainedCollectionRecord[K]]:
         )
         with self._db.query(sql) as sql_result:
             parent_keys = sql_result.scalars().all()
-        for key in parent_keys:
-            # TODO: Just in case cached records miss new parent collections.
-            # This is temporary, will replace with non-cached records soon.
-            with contextlib.suppress(KeyError):
-                yield cast(ChainedCollectionRecord[K], self._records[key])
+        # TODO: It would be more efficient to write a single query that both
+        # finds parents and all their children, but for now we do not care
+        # much about efficiency. Also the only client of this method does not
+        # need full records, only parent collection names, maybe we should
+        # change this method to return names instead.
```

Comment:
:+1: to making this return names only.

Reply:
I pushed [ceede1c](https://github.com/lsst/daf_butler/pull/904/commits/ceede1c2c838245844f5fe567a0f76b2df20a011) which changes return type to strings.